FTBFS - Debian/sid

This is probably the cause of the recent CI failure. (But we never saw the error message)

```plain
CMakeFiles/mir_umock_acceptance_tests.dir/test_libinput.cpp.o -c /home/alan/mir/tests/umock-acceptance-tests/test_libinput.cpp
In file included from /usr/src/googletest/googlemock/include/gmock/gmock.h:65,
                 from /home/alan/mir/tests/include/mir/test/doubles/mock_option.h:24,
                 from /home/alan/mir/tests/umock-acceptance-tests/test_libinput.cpp:20:
In constructor ‘testing::internal::NiceMockImpl<MockClass>::NiceMockImpl() [with Base = mir::test::doubles::MockOption]’,
    inlined from ‘testing::NiceMock<MockClass>::NiceMock() [with MockClass = mir::test::doubles::MockOption]’ at /usr/src/googletest/googlemock/include/gmock/gmock-nice-strict.h:145:26,
    inlined from ‘virtual void LibInput_DISABLED_probes_as_unsupported_without_device_access_Test::TestBody()’ at /home/alan/mir/tests/umock-acceptance-tests/test_libinput.cpp:53:31:
/usr/src/googletest/googlemock/include/gmock/gmock-nice-strict.h:112:60: error: ‘options’ may be used uninitialized [-Werror=maybe-uninitialized]
  112 |   NiceMockImpl() { ::testing::Mock::AllowUninterestingCalls(this); }
      |                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~
In file included from /usr/src/googletest/googlemock/include/gmock/gmock-function-mocker.h:42,
                 from /usr/src/googletest/googlemock/include/gmock/gmock.h:61,
                 from /home/alan/mir/tests/include/mir/test/doubles/mock_option.h:24,
                 from /home/alan/mir/tests/umock-acceptance-tests/test_libinput.cpp:20:
/usr/src/googletest/googlemock/include/gmock/gmock-spec-builders.h: In member function ‘virtual void LibInput_DISABLED_probes_as_unsupported_without_device_access_Test::TestBody()’:
/usr/src/googletest/googlemock/include/gmock/gmock-spec-builders.h:433:15: note: by argument 1 of type ‘const void*’ to ‘static void testing::Mock::AllowUninterestingCalls(const void*)’ declared here
  433 |   static void AllowUninterestingCalls(const void* mock_obj)
      |               ^~~~~~~~~~~~~~~~~~~~~~~
/home/alan/mir/tests/umock-acceptance-tests/test_libinput.cpp:53:31: note: ‘options’ declared here
   53 |     NiceMock<mtd::MockOption> options;
      |                               ^~~~~~~
cc1plus: all warnings being treated as errors
make[2]: *** [tests/umock-acceptance-tests/CMakeFiles/mir_umock_acceptance_tests.dir/build.make:79: tests/umock-acceptance-tests/CMakeFiles/mir_umock_acceptance_tests.dir/test_libinput.cpp.o] Error 1
make[2]: Leaving directory '/home/alan/mir/build-amd64'
make[1]: *** [CMakeFiles/Makefile2:4984: tests/umock-acceptance-tests/CMakeFiles/mir_umock_acceptance_tests.dir/all] Error 2
make[1]: Leaving directory '/home/alan/mir/build-amd64'
```

--- a/.github/workflows/spread.yml
+++ b/.github/workflows/spread.yml
@@ -38,7 +38,8 @@ jobs:
 
           # Only run if we have access to secrets.
           if ${{ github.event_name == 'push' || github.event.pull_request.head.repo.full_name == github.repository }}; then
-            TASKS+='"google:ubuntu-20.04-64:spread/build/sbuild:ubuntu"
+            TASKS+='"google:ubuntu-20.04-64:spread/build/sbuild:debian_sid"
+                    "google:ubuntu-20.04-64:spread/build/sbuild:ubuntu"
                     "google:ubuntu-20.04-64:spread/build/sbuild:ubuntu_hirsute"
                     "google:ubuntu-20.04-64:spread/build/sbuild:ubuntu_impish"
                     "google:ubuntu-20.04-64:spread/build/sbuild:ubuntu_arm64"
@@ -48,7 +49,6 @@ jobs:
                     "google:fedora-34-64:spread/build/fedora:amd64"'
             FAILING_TASKS='"google:ubuntu-20.04-64:spread/build/sbuild:ubuntu_devel"
                            "google:ubuntu-20.04-64:spread/build/sbuild:ubuntu_proposed"
-                           "google:ubuntu-20.04-64:spread/build/sbuild:debian_sid"
                            "google:fedora-rawhide-64:spread/build/fedora:amd64"'
           fi
         fi

--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -73,6 +73,7 @@ check_cxx_compiler_flag(-Woverloaded-virtual HAS_W_OVERLOADED_VIRTUAL)
 check_cxx_compiler_flag(-Winconsistent-missing-override HAS_W_INCONSISTENT_MISSING_OVERRIDE)
 check_cxx_compiler_flag(-Wgnu-zero-variadic-macro-arguments HAS_W_GNU_ZERO_VARIADIC_MACRO_ARGUMENTS)
 check_cxx_compiler_flag(-Wno-sign-compare HAS_W_GNU_SIGN_COMPARE)
+check_cxx_compiler_flag(-Wmaybe-uninitialized HAS_W_MAYBE_UNINITIALIZED)
 
 if (HAS_W_NULL_DEREFERENCE)
   # Avoid clang complaints about poor quality gmock/gtest headers
@@ -100,6 +101,10 @@ endif()
 if (HAS_W_GNU_SIGN_COMPARE)
   # TODO fix the code
   set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-sign-compare")
+endif()
+if (HAS_W_MAYBE_UNINITIALIZED)
+  # Avoid g++ complaints about gmock/gtest headers
+  set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-error=maybe-uninitialized")
 endif()
 
 set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fno-lto")

